contracts/igps/core/src/query.rs

```diff
@@ -68,13 +68,17 @@ pub fn quote_dispatch(
 
     let igp_message: Message = req.message.into();
 
-    let quote_res = quote_gas_payment(deps, igp_message.dest_domain, gas_limit);
+    let gas_needed = quote_gas_payment(deps, igp_message.dest_domain, gas_limit)?.gas_needed;
 
     Ok(QuoteDispatchResponse {
-        gas_amount: Some(coin(
-            quote_res?.gas_needed.to_string().parse::<u128>()?,
-            GAS_TOKEN.load(deps.storage)?,
-        )),
+        gas_amount: if gas_needed.is_zero() {
```

Comment:
nit: I'd assign to a variable on a line above to keep this easy to read, like
```
let gas_amount = if gas_needed.is_zero() {
        None
    } else {
        Some(coin(
            gas_needed.to_string().parse::<u128>()?,
            GAS_TOKEN.load(deps.storage)?,
        ))
    };
Ok(QuoteDispatchResponse {
   gas_amount
})
```

Reply:
It makes sense